lib/vecmat.gi

```diff
@@ -1411,12 +1411,12 @@ local sf, rep, ind, ind2, row, i,big,l;
   else
     if not IsField(field) then
       # not a field
-      return Immutable(matrix);
+      return matrix;
     fi;
     sf:=Size(field);
   fi;
 
-  big:=sf>256 or sf=0;
+  big:=sf>256 or sf=0 or (sf <= 256 and not field=GF(sf));
```

Comment:
I didn't find a filter that would test what I want (and thought that this is not performance critical because the GF(q)'s would only be generated for small q).
But reading the documentation of `IsFFECollection` I guess that this is a correct test to decide if the `Convert...` and `CONV...` functions are applicable in the following code.

Will change as suggested, thanks for the hint.

But the function also seems to support an integer as `field` argument. Maybe both variants are not correct?